### ISSUE TITLE
Some errors + CI fails on Laravel 11.15

### ISSUE THREAD

**Issue by damiencriado** [tpetry/laravel-postgresql-enhanced#85]
I've encountered an issue with the latest version of Laravel (11.15) during migrations.

```php
Schema::table('my_table', static function (Blueprint $table) {
    $table->decimal('my_column', 13, 4)->default(0);
    // ...
});
```

```sql
Duplicate column: 7 ERROR: column "my_column" of relation "my_table" already exists
```

The CI now fails on the latest version of Laravel: https://github.com/damiencriado/laravel-postgresql-enhanced/actions/runs/9915497290

Maybe it has something to do with this change: https://github.com/laravel/framework/pull/51373

Please let me know if you need any further information to diagnose this issue.

**Comment by hafezdivandari** [comment]
Overridden [`compileAdd`](https://github.com/tpetry/laravel-postgresql-enhanced/blob/407ca19ee36d7baeac202e66c024afbaee2b53f2/src/Schema/Grammars/GrammarTable.php#L17) and [`compileChange`](https://github.com/tpetry/laravel-postgresql-enhanced/blob/407ca19ee36d7baeac202e66c024afbaee2b53f2/src/Schema/Grammars/GrammarTypes.php#L18) methods must reflect the same changes as [`PostgresGrammar`](https://github.com/laravel/framework/pull/51373/files#diff-653f90bd1ee0a1c7c7970912484e98c8bbd30c8e9a3d2f544d53834d54e7ce7c) on PR laravel/framework#51373. Sorry for the trouble.

**Comment by tpetry** [comment]
Fixing this to work with all versions will again be tricky. I'll work on this on Monday.

**Comment by tpetry** [comment]
I've fixed the CI issues with the newest Laravel 11.x release in `0.40.1`.

But your initial issues of the column already existing should be an issue because of breaking change with Laravel 11.15.x behaviour. The issue should also exist when removing my package. From my understanding you need to reorder the schema changes in your migration.

**Comment by hafezdivandari** [comment]
Hi @tpetry, the problem is actually on the overriden `compileAdd` I think. After 11.15, we are adding columns one by one, so `$command->column` should be used instead of `$blueprint->getAddedColumns()`.

**Comment by tpetry** [comment]
`$blueprint->getAddedColumns()` is just an alias to `$command->columns` with some extra logic. So it shouldnt be the problem. And others reported the same issue with your PR.

@damiencriado Can you check whether the issue remains/disappears when removing my package?

**Comment by hafezdivandari** [comment]
> `$blueprint->getAddedColumns()` is just an alias to `$command->columns`

`$blueprint->getAddedColumns()` is an alias for `$blueprint->commands` (all added columns on blueprint), that should change to new `$command->column` (single column on the command).

**Comment by tpetry** [comment]
The example by hafezdivandari shows that indeed the current version is broken because of behavioural bc break in Laravel.

**Comment by damiencriado** [comment]
> `$blueprint->getAddedColumns()` is just an alias to `$command->columns` with some extra logic. So it shouldnt be the problem. And others reported the same issue with your PR.
> 
> @damiencriado Can you check whether the issue remains/disappears when removing my package?

Hi @tpetry ! I still have errors in my CI :(

```sql
SQLSTATE[42701]: Duplicate column: 7 ERROR:   column "my_column" of relation "my_table" already exists
```

**Comment by tpetry** [comment]
Yeah, I have now a working test case that is failing only in that specific version. Working on a fix based on the PR from @hafezdivandari.

**Comment by tpetry** [comment]
Its now finally fixed with `0.40.2`
